src/gui/advancedsettings.cpp

```diff
@@ -848,7 +851,16 @@ void AdvancedSettings::loadAdvancedSettings()
     addRow(PYTHON_EXECUTABLE_PATH, tr("Python executable path (may require restart)"), &m_pythonExecutablePath);
     // Start session paused
     m_checkBoxStartSessionPaused.setChecked(session->isStartPaused());
-    addRow(START_SESSION_PAUSED, tr("Start session in paused state"), &m_checkBoxStartSessionPaused);
+    addRow(START_SESSION_PAUSED, tr("Start BitTorrent session in paused state"), &m_checkBoxStartSessionPaused);
+    // Session shutdown timeout
+    m_spinBoxSessionShutdownTimeout.setMinimum(-1);
+    m_spinBoxSessionShutdownTimeout.setMaximum(std::numeric_limits<int>::max());
+    m_spinBoxSessionShutdownTimeout.setValue(session->shutdownTimeout());
+    m_spinBoxSessionShutdownTimeout.setSuffix(tr(" sec", " seconds"));
+    m_spinBoxSessionShutdownTimeout.setSpecialValueText(tr("-1 (unlimited)"));
+    m_spinBoxSessionShutdownTimeout.setToolTip(u"Sets the timeout for the session to be shut down gracefully, at which it will be force terminated.<br>Note that this does not apply to the saving resume data time."_s);
```

Comment:
```suggestion
    m_spinBoxSessionShutdownTimeout.setToolTip(u"Sets the timeout for the session to be shut down gracefully, at which point it will be forcibly terminated.<br>Note that this does not apply to the saving resume data time."_s);
```

Reply:
Changed.
Thank you!